OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -62,48 +67,95 @@ public Turreted(ActorInitializer init, TurretedInfo info)
 		{
 			this.info = info;
 			TurretFacing = GetInitialTurretFacing(init, info.InitialFacing);
+
+			if (info.RandomRealign)
+				randomFacing = init.Self.World.SharedRandom.Next(0, 256);
```

Comment:
I have to grab the facing each tick (or whenever it changes), do some arithmetic to make sure it is within the bounds defined, rotate it within these bounds. If the tank moves while rotating, I'd need to get another value within the limit before the rotation has finished. It would really be easier if I could do this with WAngles or proper angle support over facings.

On 2015-08-26, at 10:29 AM, Pavel Penev <notifications@github.com> wrote:

> In OpenRA.Mods.Common/Traits/Turreted.cs:
> 
> > @@ -62,48 +67,95 @@ public Turreted(ActorInitializer init, TurretedInfo info)
> >  		{
> >  			this.info = info;
> >  			TurretFacing = GetInitialTurretFacing(init, info.InitialFacing);
> > +
> > +			if (info.RandomRealign)
> > +				randomFacing = init.Self.World.SharedRandom.Next(0, 256);
> How is that any harder though
> 
> ―
> Reply to this email directly or view it on GitHub.
>